src/components/formStore/table/useFormStoreTable.tsx

```diff
@@ -420,12 +420,13 @@ export default function useFormStoreTable({
 
   React.useEffect(() => {
     if (!table.state.columnResizing.isResizingColumn) {
-      localStorage.setItem(
-        localStorageKey(form.id),
-        JSON.stringify(table.state),
-      )
+      const storageKey = localStorageKey(form.id)
+      localStorage.setItem(storageKey, JSON.stringify(table.state))
     }
-  }, [form.id, table.state])
+    // We cannot use form.id as a dependency here because it will cause table state
+    // from the previous selected form to be set in local storage for the new form
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [table.state])
```

Comment:
This may lead to similar problems in the future if we need to make more changes here later and we forget that deps are ignored. Could we instead include the `formId` in the table state somehow and compare the `form.id` and `table.state.formId` before updating local storage?